core/lib/mno_enterprise/concerns/controllers/auth/sessions_controller.rb

```diff
@@ -29,23 +29,41 @@ module ClassMethods
   # end
   
   # POST /resource/sign_in
-  # def create
-  #   super
-  # end
-  # def create
-  #   self.resource = warden.authenticate!(auth_options)
-  #   set_flash_message(:notice, :signed_in) if is_flashing_format?
-  #   sign_in(resource_name, resource)
-  #   yield resource if block_given?
-  #   respond_with resource, location: after_sign_in_path_for(resource)
-  # end
+  def create
+    @user = warden.authenticate!(auth_options)
+    if @user.requires_otp_for_login?
+      # This will trigger an after_save callback on the back end
+      # Where if there is an otp_secret then nothing will be changed
+      # But if there isn't, an unconfirmed_otp_secret will be generated.
+      @user = @user.activate_otp
```

Comment:
Should we have a check there to not send the update_attributes request to the backend if not needed?
Or is this also used to generate an `otp`?